### DISCUSSION TITLE
Dan's big Apothecary PR - 0.12.0 Status

### DISCUSSION THREAD

**Discussion by ofTheo** [openframeworks/openFrameworks#7595] [General]
Figured this would get more eyes here, but @danoli3 it would be great to get an update on the PR. 
What platforms are working what is not working etc. 

We're itching to get 0.12.0 out, as 0.11.2 is broken on some platforms so good to know if we should do 0.12.0 as is without Android or try and push and get this merged in and Android back online. 

Relevant PR: 
https://github.com/openframeworks/apothecary/pull/231

**Comment by ofTheo** [comment]
Also should mention that we don't need any x86 / 32bit libs and in terms of VS we can target 2019 or 2022 ( whichever is easier, but don't need both ).

**Comment by danoli3** [comment]
Awesome! Yeah lets get all this merged

Android side has been done for a while just need to merge in changes with latest master. 
I'll update the PR to latest. Conflict issues potential with new CMake PR 

Main issue is the project generator changes to be configured for the new Android Studio CMake System. 

Dependancy Framework Upgrades with so many Security and Fixes:

Builds working for:
Android - arm64, armv7, x86_64, x86 (NDK 24)
Windows - x86_64, x86 (Win SDK 10) - VS2019
Windows - x86_64, arm64, x86 (Win SDK 10) - VS2022
Emscripten 
(these work however newer xframework systems in progress and support for new M1 / arm64 ) 
macOS - x86_64
iOS - arm64, x86_64 simulator
tvOS - arm64, x86_64 simulator


Updated **Assimp** from 4.0.1 to **5.2.9**  - (Updated  Cmake build system)
Updated **uriparser** from 0.8.5 to **0.9.7** - (New  Cmake build system)
Updated **pugixml** from 1.9 to **1.11.4** - (New  Cmake build system)
Updated **OpenSSL** from 1.1.0h to **1.1.1u** (New  Cmake build system, +security patches)
Updated **cURL** from 7.59.0 to **8.2.0** (New  Cmake build system - fixed a bunch most platforms +security patches)
Updated **OpenCV** from 4.0.1 to 4.7.0 - (New CMake build system and fixed commands, emscripten, android, vs)
Updated **tess2** - 1.0.2 - (New Cmake build system)
Updated **FreeImage** (2) from 3.17.0 to 3.19.0 (iOS/tvOS/Android - New CMake Build system - Custom Patched core - These patches are already being used in Unix Fedora core) 
Updated **libxml2** from 2.9.4 to 2.11.4  - (New  Cmake build system)
Updated **FreeType** from 2.7 to **2.13.1** - (New Cmake build system)
Updated **uri** - (removed Boost from Android make, switched to CMake for Android)
Updated **Cairo** from 1.14.12 to 1.17.8 - (New Cmake build system)
Updated **pixman** from 0.34.0 to 0.40.0 - (New Cmake build system)
Updated **pkg-config.sh** from 0.28 to 0.29.2
Updated **json** from 3.6.1 to **3.11.2**
Updated **videoInput** with - (New Cmake build system - patched core)
Updated **GLFW** 3.3.7 to **3.3.8**  - (Updated Cmake build system)
Updated **GLEW** 1.11.0 to **2.2.0**  - (Updated Cmake build system)
Updated **LibUSB** 1.0.26 to master - (New Cmake build system)
Updated **GLM** to latest master
Updated **zlib** to 1.2.13 - - (New Cmake build system)
Updated **rtAudio** to 6.0.0 - - (New Cmake build system)
Updated **libPNG** to 1.64.0 - - (New Cmake build system)

(2): 
FreeImage 3.19.0 :
Month day, 2023 - 3.19.0
! FreeImage now uses ZLib 1.2.13
! FreeImage now uses LibPNG 1.6.39
! FreeImage now uses LibTIFF 4.5.0 (GIT patch 2023-05-18)
! FreeImage now uses LibWebP 1.2.1 (GIT patch 2021-10-29)
! FreeImage now uses LibRaw 0.21.1 (GIT patch 2023-05-13)
! FreeImage now uses LibJPEG 9d
! FreeImage now uses LibJXR 1.2 (fixed FreeImage version)
! FreeImage no longer support a 32-bit DLL or Lib version

Removal: Android Boost Dependency (file system switched to C++17)
Removal: Poco - Android Core (also fixed build script for Android if anyone wants to)
Fixed fmod download scripts 
Fixed Android Install for NDK 23 and new download links
FreeImage source code Updated to 3.19.0 for Android and patched for C++17
Removed FreeImage SubComponents (OpenEXR / LibRAWlite) removed from FreeImage util they update to support new compile targets. Updated github.com/danoli3/FreeImage to support C++17 and had to remove these broken libraries for now.
Fixed iOS Configure targets for tvOS 
Updated Docker to use CMake 3.22.1
Fixed linuxarmv7l/install.sh for new URL
Added new scripts to assist building locally (vs currently)

Apothecary can build the libraries with the following build commands:

```
# Android

./apothecary -t android -a arm64 update core
./apothecary -t android -a armv7 update core
./apothecary -t android -a x86 update core
./apothecary -t android -a x86_64 update core

# emscripten
./apothecary -t emscripten update core

# VS 
./apothecary -t vs -a arm64 update core
./apothecary -t vs -a 32 update core
./apothecary -t vs -a 64 update core

```

**Reply by dimitre** [reply]
Great news, looking forward to see it merged. 
I suppose it will need some time to incorporate and resolve conflicts, and some more time to test everything.
Is there some more important functionality that could be split in a different PR to be in a next release? 
I think GLM from the latest master is great to have.
